lisp/ess-r-xref.el

```diff
@@ -79,27 +79,32 @@ srcrefs point to temporary locations."
         (goto-char (match-beginning 0))
         (read (current-buffer))))))
 
-(defun ess-r-xref--pkg-srcfile (symbol r-src-file)
-  "Look in the source directory of the R package containing symbol SYMBOL for R-SRC-FILE."
-  (let* ((env-name (ess-string-command (format ".ess_fn_pkg(\"%s\")\n" symbol)))
-         (pkg (if (string-equal env-name "")
-                  (user-error "Can't find package for symbol %s" symbol)
-                env-name))
-         (dir (or (assoc-default pkg ess-r-xref-pkg-sources)
-                  (cond ((stringp ess-r-package-library-paths)
-                         (expand-file-name pkg ess-r-package-library-paths))
+(defun ess-r-xref--pkg-srcfile (symbol src-file &optional default-pkg)
+  "Look in the source directory of the R package containing symbol SYMBOL for SRC-FILE.
+DEFAULT-PKG is the name of the package where presumably SYMBOL is located."
+  (let* ((pkgs (delq nil
+                     (delete-dups
+                      (or (cons default-pkg
+                                (ess-get-words-from-vector (format ".ess_fn_pkg(\"%s\")\n" symbol)))
+                          (user-error "Can't find package for symbol %s" symbol)))))
+         (libdirs (cond ((stringp ess-r-package-library-paths)
+                         (list ess-r-package-library-paths))
                         ((listp ess-r-package-library-paths)
-                         (cl-loop for d in ess-r-package-library-paths
-                                  for p = (expand-file-name pkg d)
-                                  when (file-exists-p p) return p))
-                        (t (user-error "Invalid value of `ess-r-package-library-paths'")))))
-         (file (when dir (expand-file-name r-src-file dir))))
+                         ess-r-package-library-paths)
+                        (t (user-error "Invalid value of `ess-r-package-library-paths'"))))
+         (pkg.dir (or (when (= (length pkgs) 1)
```

Comment:
Symbol separator:

```suggestion
         (pkg-dir (or (when (= (length pkgs) 1)
```

Reply:
This is intentional, it's a cons cel `(pkg . dir)` this is why the dot.

Reply:
I'm not a fan of this convention because it makes the code look messy, also it's the first time I see this. If it is important to mention it's a cell, maybe `cell-pkg-dir`? Otherwise I'd just use `pkg-info` or something like that.

Reply:
Well, style is 80% what you are used to. I think we should use letters to full extent to convey the semantic meaning of the name, not jus stubbornly stick to `-` or `_` etc. Not a big deal though, I can call it `loc` or something. This `car.cdr` it's my own thing. I think I used it in ESS as well.

Reply:
In my opinion it's unemacsy to use special symbols in names, like `?` for predicates etc. Maybe we should discuss this separately and use the conventional style for now? It's certainly easier to just use Emacs core conventions, when they exist.

Reply:
I surely understand where you are coming from and that "." does iritate me for the same reasons. I will change it to `loc`. Still I think it's a matter of getting used to.